datafusion/common/src/join_type.rs

```diff
@@ -113,6 +118,9 @@ pub enum JoinSide {
     Left,
     /// Right side of the join
     Right,
+    /// Neither side of the join, used for Mark joins where the mark column does not belong to
```

Comment:
Is there such a thing as right mark join? If so, can we add a issue/TODO for it?

Reply:
There is. I started adding both in a single PR but realised that is was more changes then I expected, so I did LeftMark first to keep the PR smaller. But created an issue for RightMark here: https://github.com/apache/datafusion/issues/13138